osu.Game/Overlays/SkinEditor/SkinEditorOverlay.cs

```diff
@@ -182,5 +201,30 @@ private void setTarget(OsuScreen? target)
                 skinEditor = null;
             }
         }
+
+        private readonly Bindable<bool> beatmapSkins = new Bindable<bool>();
+        private bool beatmapSkinsOriginalState;
+
+        private void globallyDisableBeatmapSkinSetting()
+        {
+            if (beatmapSkins.Disabled)
+                return;
+
+            // The skin editor doesn't work well if beatmap skins are being applied to the player screen.
+            // To keep things simple, disable the setting game-wide while using the skin editor.
+            beatmapSkinsOriginalState = beatmapSkins.Value;
+
+            beatmapSkins.Value = false;
+            beatmapSkins.Disabled = true;
```

Comment:
Is there a reason why this is essentially manually reimplementing what a bindable lease is?

```diff
diff --git a/osu.Game/Overlays/SkinEditor/SkinEditorOverlay.cs b/osu.Game/Overlays/SkinEditor/SkinEditorOverlay.cs
index b120faa45f..4f0028de64 100644
--- a/osu.Game/Overlays/SkinEditor/SkinEditorOverlay.cs
+++ b/osu.Game/Overlays/SkinEditor/SkinEditorOverlay.cs
@@ -203,7 +203,7 @@ private void setTarget(OsuScreen? target)
         }
 
         private readonly Bindable<bool> beatmapSkins = new Bindable<bool>();
-        private bool beatmapSkinsOriginalState;
+        private LeasedBindable<bool>? leasedBeatmapSkins;
 
         private void globallyDisableBeatmapSkinSetting()
         {
@@ -212,19 +212,14 @@ private void globallyDisableBeatmapSkinSetting()
 
             // The skin editor doesn't work well if beatmap skins are being applied to the player screen.
             // To keep things simple, disable the setting game-wide while using the skin editor.
-            beatmapSkinsOriginalState = beatmapSkins.Value;
-
-            beatmapSkins.Value = false;
-            beatmapSkins.Disabled = true;
+            leasedBeatmapSkins = beatmapSkins.BeginLease(true);
+            leasedBeatmapSkins.Value = false;
         }
 
         private void globallyReenableBeatmapSkinSetting()
         {
-            if (!beatmapSkins.Disabled)
-                return;
-
-            beatmapSkins.Disabled = false;
-            beatmapSkins.Value = beatmapSkinsOriginalState;
+            leasedBeatmapSkins?.Return();
+            leasedBeatmapSkins = null;
         }
     }
 }

```